plugins/plugin-chart-echarts/src/components/Echart.tsx

```diff
@@ -26,38 +26,54 @@ const Styles = styled.div<EchartsStylesProps>`
   width: ${({ width }) => width};
 `;
 
+const usePrevious = <T extends unknown>(value: T): T | undefined => {
+  const ref = useRef<T>();
+  useEffect(() => {
+    ref.current = value;
+  });
+  return ref.current;
+};
+
 export default function Echart({
   width,
   height,
   echartOptions,
   eventHandlers,
-  selectedValues,
+  selectedValues = {},
 }: EchartsProps) {
   const divRef = useRef<HTMLDivElement>(null);
   const chartRef = useRef<ECharts>();
+  const currentSelection = Object.keys(selectedValues) || [];
+  const previousSelection = usePrevious<string[]>(currentSelection) || [];
```

Comment:
I think `usePrevious` can already infer types from `currentSelection`?

Reply:
Can we maybe just always downplay all previous highlighted selection and highlight all current selection?

```ts
const previousSelection = useRef(currentSelection);

useEffect(() => {
  if (previousSelection.current.length) {
    // downplay
  }
  if (currentSelection.length) {
    // highlight
  }
  previousSelection.current = currentSelection;
});
```

Reply:
I initially did that, but as I didn't know how expensive the operation is, so I erred on the side of caution. But I'll change it and if it does turn out to become a bottleneck.

Reply:
I changed it so that we highlight all currently selected values, but only downplay the previously selected values that aren't currently selected to avoid flipflopping.